bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/airbase/AirbaseEnums.java

```diff
@@ -51,39 +56,73 @@ public static AirbaseMode fromValue(int value) {
 
             LOGGER.debug("Unexpected Mode value of \"{}\"", value);
 
-            // Default to auto
-            return AUTO;
+            // Default to cold
+            return COLD;
         }
     }
 
     public enum AirbaseFanSpeed {
-        AUTO("A"),
-        LEVEL_1("1"),
-        LEVEL_2("3"),
-        LEVEL_3("5");
+        // level,f_auto,f_airside
+        LEVEL_1      (1, false, false),
+        LEVEL_2      (2, false, false),
+        LEVEL_3      (3, false, false),
+        LEVEL_4      (4, false, false),
+        LEVEL_5      (5, false, false),
+        AUTO_LEVEL_1 (1, true, false),
+        AUTO_LEVEL_2 (2, true, false),
+        AUTO_LEVEL_3 (3, true, false),
+        AUTO_LEVEL_4 (4, true, false),
+        AUTO_LEVEL_5 (5, true, false),
+        AIRSIDE      (1, false, true);
 
         private static final Logger LOGGER = LoggerFactory.getLogger(AirbaseFanSpeed.class);
-        private final String value;
+        private final int level;
+        private final boolean auto;
+        private final boolean airside;
+
+        AirbaseFanSpeed(int level, boolean auto, boolean airside) {
+            this.level = level;
+            this.auto = auto;
+            this.airside = airside;
+        }
 
-        AirbaseFanSpeed(String value) {
-            this.value = value;
+        public int getLevel() { 
+            return level;
         }
 
-        public String getValue() {
-            return value;
+        public boolean getAuto() {
+            return auto;
+        }
+
+        public boolean getAirside() {
+            return airside;
+        }
+
+        public String getLabel() {
+            if (airside) {
+                return "Airside";
+            }
+            String label = "";
+            if (auto) {
+                label = "Auto ";
+            }
+            return label + "Level " + Integer.toString(level);
         }
 
-        public static AirbaseFanSpeed fromValue(String value) {
+        public static AirbaseFanSpeed fromValue(int rate, boolean auto, boolean airside) { // convert from f_rate, f_auto, f_airside
+            if (airside) {
+                return AIRSIDE;
+            }
             for (AirbaseFanSpeed m : AirbaseFanSpeed.values()) {
-                if (m.getValue().equals(value)) {
+                if (m.getLevel() == rate && m.getAuto() == auto && m.getAirside() == airside) {
                     return m;
                 }
             }
 
-            LOGGER.debug("Unexpected FanSpeed value of \"{}\"", value);
+            LOGGER.debug("Unexpected FanSpeed value from rate={}, auto={}, airside={}", rate, auto, airside);
```

Comment:
I still think this is wrong.You should throw `IllegalArgumentException` and ctah that wherever it is used. If needed, you can use a default there.

Reply:
The data comes from the device. It isn't something that rules will assign to. So the error would happen if say the device returns a combination of values that aren't covered / supported by this binding. Would `IllegalArgumentException` still be the correct exception type?

Reply:
This logging/default value is the convention in this binding from the original developer, and it is done everywhere.